happi/tests/test_cli.py

```diff
@@ -87,6 +99,21 @@ def test_search(happi_cfg):
     assert [r.device for r in res] == [r.device for r in res_cli]
 
 
+def test_search_with_name(happi_cfg, caplog):
+    client = happi.client.Client.from_config(cfg=happi_cfg)
+    res = client.search_regex(name='TST_BASE_PIM2')
+    res_cli = happi.cli.happi_cli(['--verbose', '--path', happi_cfg, 'search',
+                                   'TST_BASE_PIM2'])
+    assert [r.device for r in res] == [r.device for r in res_cli]
+    # test duplicate search parameters
+    with caplog.at_level(logging.ERROR):
+        res_cli = happi.cli.happi_cli(['--verbose', '--path', happi_cfg,
+                                       'search', 'TST_BASE_PIM2',
+                                       'name=TST_BASE_PIM2'])
+        assert "Received duplicate search criteria" in caplog.text
```

Comment:
This is less of a problem than the one above since we control the output 100%, but in general I think it would be better to set up some return codes in `happi_cli` to check which branch we went down rather than relying on log messages.

This is more of a general thought, editing the return structure of the cli file to make the testing smoother doesn't have to be in scope for this PR.

Reply:
Yeah I think this should be solved with refactoring the `cli.py` module to allow for that.....

Reply:
To be fair, in the issue, we did say "please help us write tests to get coverage" and not "refactor, break up cli.py, and make tests".

I think better communication between our groups from both sides could have saved you a bit of time here. We're always up for chatting on Slack, so open up a thread with us for future efforts and we can discuss ideas/concerns/thoughts/etc.

Reply:
Yeah it seems like overall I had the wrong approach with this PR, since pretty much I am checking against log messages for most of the tests that I added... maybe a better approach would have been to have an Issue that refactors the `cli` module first and then have an Issue for testing it...

Reply:
I think I didn't realize how awkward the whole cli.py was architected in the first place. That's probably my fault. It doesn't help that I'm getting repeatedly pulled into looking at other projects and am having trouble keeping up here...

Reply:
No judgement intended, @ZLLentz, sorry if it sounded that way! Reiterating my usual offer, let me know if I can help with the other requests.

Reply:
@ZLLentz same here, I did not mean any judgement on anybody except myself haha, It was my bad, I think I should have talked to you on how we want to approach this PR, and that would have been  much better